src/golang/lib/collections/workflow_dag_result/sqlite.go

```diff
@@ -51,17 +50,22 @@ func (r *sqliteReaderImpl) GetKOffsetWorkflowDagResultsByWorkflowId(
 func (w *sqliteWriterImpl) CreateWorkflowDagResult(
 	ctx context.Context,
 	workflowDagId uuid.UUID,
+	execState *shared.ExecutionState,
 	db database.Database,
 ) (*WorkflowDagResult, error) {
-	insertColumns := []string{IdColumn, WorkflowDagIdColumn, StatusColumn, CreatedAtColumn}
+	insertColumns := []string{IdColumn, WorkflowDagIdColumn, StatusColumn, CreatedAtColumn, ExecStateColumn}
 	insertWorkflowDagResultStmt := db.PrepareInsertWithReturnAllStmt(tableName, insertColumns, allColumns())
 
 	id, err := utils.GenerateUniqueUUID(ctx, tableName, db)
 	if err != nil {
 		return nil, err
 	}
 
-	args := []interface{}{id, workflowDagId, shared.PendingExecutionStatus, time.Now()}
+	if execState.Timestamps == nil || execState.Timestamps.PendingAt == nil {
```

Comment:
This should be commented somewhere that the pendingat timestamp can't be nil.

Reply:
Added a few comments in the interface struct. Also created https://linear.app/aqueducthq/issue/ENG-1701/deprecate-createdat-field-for-workflow-dag-results to remove this constraints once that is done.